### PR TITLE
Improve metric names and use GaugeVecs for some things

### PR DESCRIPTION
**Pull request by expenses**
@mxinden.

### CODE REVIEW

**Comment by nodebreaker0-0** [comment]
This suggestion is very good, very useful for building extended metrics later.

**Review by mxinden — COMMENTED** [review]
This pull request removes the following metrics:

```rust
	finality_height: Gauge<U64>,
	best_height: Gauge<U64>,
	peers_num: Gauge<U64>,
	tx_count: Gauge<U64>,
	node_memory: Gauge<U64>,
	node_cpu: Gauge<F64>,
	node_download: Gauge<U64>,
	node_upload: Gauge<U64>,
	sync_target: Gauge<U64>,
```

Is this intentional? Am I missing something? Did you forget to push a commit?

**Comment by expenses** [comment]
@mxinden no, it replaces them with vecs.

**Comment by expenses** [comment]
so now instead of `finality_height`, its `block_height{number = "finality"}`. I'll change it to be more like https://github.com/paritytech/substrate/pull/4511#discussion_r370725547 though.

File: client/service/src/builder.rs [hunk 1]
```diff
@@ -1087,16 +1059,19 @@ ServiceBuilder<
...
-				metrics.node_cpu.set(f64::from(cpu_usage));
-				metrics.tx_count.set(txpool_status.ready as u64);
-				metrics.finality_height.set(finalized_number);
-				metrics.best_height.set(best_number);
-				metrics.peers_num.set(num_peers as u64);
-				metrics.node_download.set(net_status.average_download_per_sec);
-				metrics.node_upload.set(net_status.average_upload_per_sec);
+				metrics.memory_usage_bytes.set(memory);
+				metrics.cpu_usage_percentage.set(f64::from(cpu_usage));
+				metrics.transactions_total.inc_by(txpool_status.ready as u64);
```
Comment (mxinden) [inline client/service/src/builder.rs]: If I am not mistaken [txpool_status.ready](https://github.com/paritytech/substrate/blob/master/primitives/transaction-pool/src/pool.rs#L42) is not being reset after this call, right? If we have two scrapes and thus call `metrics.transactions_total.inc_by(txpool_status.ready as u64);` twice without a reset, the number will not be correct.

Comment (mxinden) [reply client/service/src/builder.rs]: In addition the `Number of transactions in the ready queue` is not a monotonically increasing number. Thus a counter would not be correct, but a gauge would need to be used.

Comment (mxinden) [reply client/service/src/builder.rs]: Also does this really represent the `"Total number of transactions"` or just the number of transactions that are currently in the ready queue?

```rust
	/// Number of transactions in the ready queue.
	pub ready: usize,
```

**Review by mxinden — APPROVED** [review]
Thanks for cleaning this up @expenses! I left a couple of comments. Otherwise this looks good to me.

File: client/service/src/builder.rs (hunk 1 above)
Comment (expenses) [reply client/service/src/builder.rs]: Hmm, yeah I think just a normal gauge would be best then.

Comment (expenses) [reply client/service/src/builder.rs]: Done

**Comment by expenses** [comment]
@nodebreaker0-0 think we're good to merge this.

**Comment by nodebreaker0-0** [comment]
Merge complete

**Review by mxinden — COMMENTED** [review]
A bit late. Sorry. Looks good to me @expenses.

**Pull request merged**